src/manager.cpp

```diff
@@ -627,10 +627,68 @@ int Manager::updateKeywordOnDbus(
     const types::Path& i_inventoryObjPath, const types::Path& i_fruPath,
     const types::WriteVpdParams i_paramsToWriteData)
 {
-    (void)i_inventoryObjPath;
-    (void)i_fruPath;
-    (void)i_paramsToWriteData;
-
+    try
+    {
+        // Extract data from input
+        if (const types::IpzData* l_ipzData =
+                std::get_if<types::IpzData>(&i_paramsToWriteData))
+        {
+            const types::Record& l_recordName = std::get<0>(*l_ipzData);
+            const types::DbusInterface& l_interfaceName = constants::ipzVpdInf +
+                                                          l_recordName;
+            const types::DbusPropertyName& l_propertyName =
+                std::get<1>(*l_ipzData);
+
+            // Read the keyword's value from hardware to write the same on
+            // D-bus.
+            const types::DbusVariantType l_keywordValue = readKeyword(
+                i_fruPath, types::ReadVpdParams(
+                               std::make_tuple(l_recordName, l_propertyName)));
+
+            if (updateKeywordViaPIM(i_inventoryObjPath, l_interfaceName,
+                                    l_propertyName, l_keywordValue))
+            {
+                if (const types::BinaryVector* l_kwValue =
+                        std::get_if<types::BinaryVector>(&l_keywordValue))
+                {
+                    return l_kwValue->size();
+                }
+            }
+        }
+    }
+    catch (const std::exception& l_exception)
+    {
+        // TODO: Log PEL
+    }
     return -1;
 }
+
+bool Manager::updateKeywordViaPIM(const types::Path& i_inventoryObjPath,
+                                  const types::DbusInterface& i_interfaceName,
+                                  const types::DbusPropertyName& i_propertyName,
+                                  const types::DbusVariantType& i_propertyValue)
+{
+    // Create property map which is required to form D-bus object map
+    types::PropertyMap l_propertyMap = {
```

Comment:
If "updateKeywordOnDbus" on DBus is restricted to update only on DBus then here just data type is being created and value is packed in that. This can be done in place.
I don't see a need for a new function here.

Reply:
done